pkg/controller/uniteddeployment/adapter/statefulset_adapter.go

```diff
@@ -0,0 +1,260 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	alpha1 "github.com/openkruise/kruise/pkg/apis/apps/v1alpha1"
+	kruisectlutil "github.com/openkruise/kruise/pkg/controller/util"
+	"github.com/openkruise/kruise/pkg/util/refmanager"
+)
+
+type StatefulSetAdapter struct {
+	client.Client
+
+	Scheme *runtime.Scheme
+}
+
+// NewResourceObject creates a empty StatefulSet object.
+func (a *StatefulSetAdapter) NewResourceObject() runtime.Object {
+	return &appsv1.StatefulSet{}
+}
+
+// NewResourceListObject creates a empty StatefulSetList object.
+func (a *StatefulSetAdapter) NewResourceListObject() runtime.Object {
+	return &appsv1.StatefulSetList{}
+}
+
+// GetObjectMeta returns the ObjectMeta of the subset of StatefulSet.
+func (a *StatefulSetAdapter) GetObjectMeta(obj metav1.Object) *metav1.ObjectMeta {
```

Comment:
Can we try not expose this as interface?

Reply:
done